ipykernel/ipkernel.py

```diff
@@ -274,21 +274,22 @@ def do_complete(self, code, cursor_pos):
 
     def _experimental_do_complete(self, code, cursor_pos):
         """
-        Experimental completions from IPython, using Jedi. 
+        Experimental completions from IPython, using Jedi.
         """
         if cursor_pos is None:
             cursor_pos = len(code)
         with provisionalcompleter():
             raw_completions = self.shell.Completer.completions(code, cursor_pos)
             completions = list(rectify_completions(code, raw_completions))
-            
+
             comps = []
             for comp in completions:
                 comps.append(dict(
                             start=comp.start,
                             end=comp.end,
                             text=comp.text,
                             type=comp.type,
+                            signature=getattr(comp, 'signature', '')
```

Comment:
Depending on the IPython version, completions may or may not return the function signature. Therefor I used `getattr` to make sure we stay compatible.